### ISSUE TITLE
Add new data structures and algorithms

### ISSUE THREAD

**Issue by nb2998** [nb2998/DS_Algo#141]
Codes are accepted in all languages (if not present already). Add them in a new folder if the folder doesn't exist already.

Any spam pull request will be marked as spam and will not be counted as a contribution towards hacktoberfest.

**Comment by ayushhj** [comment]
@nb2998 Hi mam, I want to contibute to this repo by adding some algorithms as a  part of hacktoberfest. Can i do so?

**Comment by ayushi-1408** [comment]
Hey! I would love to contribute on this issue.

**Comment by nb2998** [comment]
@ayushhj @ayushi-1408 Sure, go ahead. Make sure you read the guidelines in Contributing.md before starting

**Comment by kavyamehra20** [comment]
please assign this repo to me

**Comment by parjanyaacoder** [comment]
@nb2998 Pls assign this issue to me.

**Comment by nb2998** [comment]
Multiple people can take up the issue, just mention the algorithm you will be working on :)

**Comment by kavyamehra20** [comment]
I have added array reversing in cpp

**Comment by Renu-code** [comment]
Hello, can I add sorting algorithms??

**Comment by saitarunmandiga** [comment]
hello can I add Linear Probing Hash Table data and Quadratic Probing Hash Table

**Comment by parjanyaacoder** [comment]
@nb2998  I want to work on Pancake Sort in Java.

**Comment by Hinal-Srivastava** [comment]
Hi!
Can you assign me to this? I'll like to add Shell Sort

**Comment by Anjali2001saxena** [comment]
Hi!
Can I add bucket sort to this repo?

**Comment by Anjali2001saxena** [comment]
Hi!
Can I add count sort in java?

**Comment by Heisenberg-Vardan** [comment]
Hi @nb2998 I would like to add a few basic examples of dynamic programming

**Comment by iamraghs** [comment]
can i solve dp problem..

**Comment by SaiSrichandra** [comment]
I want to add Heap Sort in java

**Comment by SaiSrichandra** [comment]
Hi @nb2998 I would like to add Bucket Sort in C++
Please assign it to me

**Comment by ayushi-1408** [comment]
Hey! @nb2998  I have added Selection and insertion sort using C language. Please have a look!!
Happy Coding!!

**Comment by SaiSrichandra** [comment]
#164 
Hello! @nb2998 
I have added Bucket Sort in C++
Please check it out

**Comment by sreenidhi3** [comment]
Hey! I would love to contribute by adding a Backtracking folder with Sudoku Implementation in java.

**Comment by AnkitT0204** [comment]
can I add a 2d array shell rotation question?

**Comment by parthchainani** [comment]
can i add dynamic and staic queue implementation in cpp

**Comment by hryxna** [comment]
hey @nb2998, I would like to open a new issue with Luhn's Algorithm as it's quite a unique one.
feel free to check it out 😄.
https://github.com/hryxna/Java-Algos/tree/master/LuhnsAlgo

Also, I have made an issue with another guy as well in his repo do check-out that too
https://github.com/aditya109/git-osp-for-beginners/issues/178
